improver/calibration/beta_recalibration.py

```diff
@@ -0,0 +1,94 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Module containing class for recalibrating blended probabilities."""
+
+from typing import Any, Dict, Optional
+
+import cf_units
+import iris
+import numpy as np
+from iris.exceptions import CoordinateNotFoundError
+from scipy.stats import beta
+
+from improver import PostProcessingPlugin
+from improver.metadata.probabilistic import is_probability
+
+
+class BetaRecalibrate(PostProcessingPlugin):
+    """Recalibrate probabilities using the cumulative distribution function
+    of the beta distribution.
+    """
+
+    def __init__(self, recalibration_dict: Optional[Dict[str, Any]] = None):
+        """
+        Args:
+            recalibration_dict:
+                Dictionary from which to calculate alpha and beta parameters for
+                recalibrating blended output using the beta distribution. Dictionary
+                format is as specified below. Weights will be interpolated over the
+                forecast period from the values specified in the dictionary.
+
+        Recalibration dictionary format::
+
+            {
+                "forecast_period": [7, 12],
+                "alpha": [1, 1.5],
+                "beta": [1.3, 2],
+                "units": "hours",
+            }
+
+        """
+        self.recalibration_dict = recalibration_dict
+
+    def process(self, cube):
+        """Recalibrate cube using the beta distribution with the alpha
+        and beta parameters specified in self.recalibration_dict.
+
+        Args:
+            cube:
+                A cube containing a forecast_period coordinate.
+
+        Returns:
+            A cube having the same dimensions as the input, with data
+            transformed by the beta distribution cdf.
+
+        Raises:
+            CoordinateNotFoundError: if cube does not contain probability data
+            CoordinateNotFoundError: if cube does not contain forecast_period coordinate
+            RuntimeError: if any interpolated values of alpha or beta are <= 0
+        """
+
+        if not (is_probability(cube)):
+            raise CoordinateNotFoundError(
+                "Input cube must be a probability forecast "
+                "and contain a threshold coordinate."
+            )
+        if len(cube.coords("forecast_period")) == 0:
+            raise CoordinateNotFoundError(
```

Comment:
Code coverage is flagging a couple of exceptions here that are not tested.

Reply:
added new tests for these cases